### ISSUE TITLE
surround with tag does not work with dot inside tag name

### ISSUE THREAD

**Issue by deviantfero** [emacs-evil/evil-surround#155]
for example, if I try to surround `<div></div>` with `<Table.Content>`, the following will be the result

```
<Table.Content>
  <div></div>
<Table/>
```

the expected result would be

```
<Table.Content>
  <div></div>
<Table.Content/>
```

**Comment by ninrod** [comment]
Yes, I think that it would be nice to behave like you describe. 

on the other hand, not to be nitpicky, but [the spec](https://www.w3.org/TR/2011/WD-html-markup-20110405/syntax.html#syntax-elements) says that: "Tag names are used within element start tags and end tags to give the element’s name. HTML elements all have names that only use characters in the range 0–9, a–z, and A–Z". 

Is this name you are using allowed? does it really work?

**Comment by deviantfero** [comment]
It is used all over in modern JSX libraries like react, here's a example on a couple of frameworks (there should be code samples in those URLs):

* https://react.semantic-ui.com/collections/table/
* https://ant.design/components/menu/

It's not a deal-breaker, but it would be really convenient to have this supported

**Comment by ninrod** [comment]
Yes, I see. It would be really useful.

**Comment by ninrod** [comment]
Solved through @deviantfero's PR (#156) in d210e1fc2cf1c2d095471cefa700a0d1703f4ab6
